AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/src/Keyrings/AwsKms/AwsKmsHierarchicalKeyring.dfy

```diff
@@ -391,28 +405,26 @@ module AwsKmsHierarchicalKeyring {
       requires cryptoPrimitives.ValidState()
       modifies cryptoPrimitives.Modifies
       ensures cryptoPrimitives.ValidState()
-      ensures cacheId.Success? ==> |cacheId.value| == 32
+      ensures cacheId.Success? ==> |cacheId.value| == 48
     {
       :- Need(
         && UTF8.Decode(branchKeyIdUtf8).MapFailure(WrapStringToError).Success?
         && var branchKeyId := UTF8.Decode(branchKeyIdUtf8).MapFailure(WrapStringToError).value;
         && 0 <= |branchKeyId| < UINT32_LIMIT,
         E("Invalid Branch Key ID Length")
       );
+      var hashAlgorithm := Crypto.DigestAlgorithm.SHA_384;
 
-      var branchKeyId := UTF8.Decode(branchKeyIdUtf8).value;
-      var lenBranchKey := UInt.UInt32ToSeq(|branchKeyId| as uint32);
+      // Resource: Hierarchy Keyring [0x01]
```

Comment:
Blocking/Issue:
I wrote Keyring `0x02` and CMM `0x01`, it is flipped here.
Strictly speaking, this will not cause an error, as long as we are consistent.
But drifting from the Spec/Design in this way,
and failing to record it/revising the spec/design,
is an easy way to cause issues tomorrow.

This is why we SHOULD put the function to calculate the Cache formula in `AwsCryptographicMaterialProviders/src/CMCs`.

For now, even if you do not want to put the formula calculation in `CMCs`,
I am going to block on defining the constants here.

Create `AwsCryptographicMaterialProviders/src/CMCs/CacheIdentifiers.dfy`,
and define the constants there.

Then import and use them here.

By Constants, I mean:
- Null Byte : `0x00`
- ResourceID 
- ScopeID